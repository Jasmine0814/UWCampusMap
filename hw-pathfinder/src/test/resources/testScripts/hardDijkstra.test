# find the shortest path with Dijkstra's algorithm

CreateGraph Graph1

AddNode Graph1 A
AddNode Graph1 B
AddNode Graph1 C
AddNode Graph1 D
AddNode Graph1 E

AddEdge Graph1 A B 2.5
AddEdge Graph1 B C 7.774477
AddEdge Graph1 C D 1.5
AddEdge Graph1 D E 1.5
AddEdge Graph1 E A 6.0
AddEdge Graph1 D C 2.0
AddEdge Graph1 E D 2.0
AddEdge Graph1 D A 9.0
AddEdge Graph1 B D 1.21111

FindPath Graph1 A E
FindPath Graph1 E C

