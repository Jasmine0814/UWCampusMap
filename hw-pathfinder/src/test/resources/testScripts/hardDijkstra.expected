# find the shortest path with Dijkstra's algorithm

created graph Graph1

added node A to Graph1
added node B to Graph1
added node C to Graph1
added node D to Graph1
added node E to Graph1

added edge 2.500 from A to B in Graph1
added edge 7.774 from B to C in Graph1
added edge 1.500 from C to D in Graph1
added edge 1.500 from D to E in Graph1
added edge 6.000 from E to A in Graph1
added edge 2.000 from D to C in Graph1
added edge 2.000 from E to D in Graph1
added edge 9.000 from D to A in Graph1
added edge 1.211 from B to D in Graph1

path from A to E:
A to B with weight 2.500
B to D with weight 1.211
D to E with weight 1.500
total cost: 5.211
path from E to C:
E to D with weight 2.000
D to C with weight 2.000
total cost: 4.000
